www/jury/balloons.php

```diff
@@ -52,6 +85,62 @@
     addSubmit($viewall ? 'view unsent only' : 'view all') . "</p>\n" .
     addEndForm();
 
+
+// Filtering by affiliation or room
+$affils = $DB->q('TABLE SELECT affilid,
```

Comment:
I think this query currently does not return what it should: the lists of all affiliations and rooms of teams that are part of selected list of contests (which can be one, or all active contests). Some issues:
- the array variable `$contestids` (currently defined down in the code) should be used instead of `$cdata['cid']`
- I think the where clause `c.cid = %i` is superfluous
- the generated filter list contains duplicate universities (not sure why)

I'd suggest to select from `team` and then LEFT JOIN on `team_affiliation`; the only point is then that empty affiliations have to be filtered out, or that would show up in the filter list and be selected by default.